### PR TITLE
[feat:4107] Mysql connection security check extract utils

### PR DESCRIPTION
**Pull request by aiceflower**
Mysql connection security check extract common utils https://github.com/apache/linkis/issues/4107

### CODE REVIEW

File: linkis-commons/linkis-common/src/main/java/org/apache/linkis/common/utils/SecurityUtils.java [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+    }
+
+    Iterator<Map.Entry<String, Object>> iterator = paramsMap.entrySet().iterator();
+    while (iterator.hasNext()) {
+      Map.Entry<String, Object> entry = iterator.next();
+      String key = entry.getKey();
+      Object value = entry.getValue();
+      if (StringUtils.isBlank(key) || value == null || StringUtils.isBlank(value.toString())) {
+        logger.warn("Invalid parameter key or value is blank.");
+        iterator.remove();
```
Comment (peacewong) [inline linkis-commons/linkis-common/src/main/java/org/apache/linkis/common/utils/SecurityUtils.java:137]: Why remove?

File: linkis-commons/linkis-common/src/main/java/org/apache/linkis/common/utils/SecurityUtils.java [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+      url = url + AND_SYMBOL + extraParamString;
+    }
+    return url;
+  }
+
+  public static void appendMysqlForceParams(Map<String, Object> extraParams) {
+    extraParams.putAll(parseMysqlUrlParamsToMap(MYSQL_FORCE_PARAMS.getValue()));
+  }
+
+  public static String checkJdbcSecurity(String url) {
```
Comment (peacewong) [inline linkis-commons/linkis-common/src/main/java/org/apache/linkis/common/utils/SecurityUtils.java:92]: checkJdbcSecurity should clearSensitiveParams

**Review by peacewong — DISMISSED** [review]

File: linkis-commons/linkis-common/src/main/java/org/apache/linkis/common/utils/SecurityUtils.java (hunk 1 above)
Comment (aiceflower) [reply linkis-commons/linkis-common/src/main/java/org/apache/linkis/common/utils/SecurityUtils.java:137]: Get rid of arguments where key or value is blank.

File: linkis-commons/linkis-common/src/main/java/org/apache/linkis/common/utils/SecurityUtils.java (hunk 2 above)
Comment (aiceflower) [reply linkis-commons/linkis-common/src/main/java/org/apache/linkis/common/utils/SecurityUtils.java:92]: Throw an exception directly if there are sensitive parameters.

File: linkis-commons/linkis-common/src/main/java/org/apache/linkis/common/exception/LinkisSecurityException.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.linkis.common.exception;
+
+public class LinkisSecurityException extends LinkisRuntimeException {
+  LinkisSecurityException(int errCode, String desc, String ip, int port, String serviceKind) {
+    super(errCode, desc, ip, port, serviceKind);
```
Comment (casionone) [inline linkis-commons/linkis-common/src/main/java/org/apache/linkis/common/exception/LinkisSecurityException.java:22]: unused, can be removed

**Review by casionone — APPROVED** [review]

**Pull request merged**